libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/roadname/api/MapboxRoadNameLabelApi.kt

```diff
@@ -1,25 +1,70 @@
 package com.mapbox.navigation.ui.maps.roadname.api
 
+import com.mapbox.api.directions.v5.models.BannerComponents
 import com.mapbox.navigation.base.road.model.Road
-import com.mapbox.navigation.ui.maps.roadname.RoadNameAction
-import com.mapbox.navigation.ui.maps.roadname.RoadNameProcessor
-import com.mapbox.navigation.ui.maps.roadname.RoadNameResult
 import com.mapbox.navigation.ui.maps.roadname.model.RoadLabel
+import com.mapbox.navigation.ui.shield.api.MapboxRouteShieldApi
+import com.mapbox.navigation.ui.shield.model.RouteShieldCallback
 
 /**
  * The API allows you to generate data to be rendered as a part of road name labels.
  */
-class MapboxRoadNameLabelApi {
+class MapboxRoadNameLabelApi constructor(
+    private val routeShieldApi: MapboxRouteShieldApi = MapboxRouteShieldApi()
+) {
 
     /**
      * The method takes in [Road] and uses the model to extract the road name, shield name and
      * request the route shield based on the [Road.shieldUrl].
      * @param road Road
      * @return RoadLabel
      */
+    @Deprecated(
+        message = "The API is redundant. To render a road name instantiate the MapboxRoadNameView" +
+            "and invoke renderRoadName by passing Road object that can be obtained via location" +
+            "observer using onNewLocationMatcherResult. The method will not be supported and " +
+            "will be removed.",
+        replaceWith = ReplaceWith("")
+    )
     fun getRoadNameLabel(road: Road): RoadLabel {
-        val action = RoadNameAction.GetRoadNameLabel(road)
-        val result = RoadNameProcessor.process(action) as RoadNameResult.RoadNameLabel
-        return RoadLabel(result.name, result.shield, result.shieldName)
+        return RoadLabel(road.name, null, road.shieldName)
+    }
+
+    /**
+     * Given the current [Road] the method requests legacy road shields (if available) using
```

Comment:
Is "legacy" a way that we want to call the "old" shields (not mapbox design ones) publicly? What's the preference of the Navigation APIs team?

cc @browndp08